### PR TITLE
API/UCT: Sockaddr calls - server callback and uct_ep_create_sockaddr.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
@yosefe @shamisp @bbenton Please review.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4869/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -1318,6 +1313,12 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
  *                               remote peer.
  * @param [in]  length           Length of the private data.
  * @param [out] ep_p             Handle to the created endpoint.
+ *
+ * @return UCS_INPROGRESS        The connection to the remote peer was initiated.
```
Comment (yosefe) [inline src/uct/api/uct.h]: pls document what would UCS_OK mean (at least in theory)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4878/ for details (Mellanox internal link).

File: src/uct/api/uct.h (hunk 1 above)
Comment (alinask) [reply src/uct/api/uct.h]: You mean for transports that can finish the connection establishment on the client side 'in place'? For which the callback won't be invoked?

Comment (yosefe) [reply src/uct/api/uct.h]: yes

**Comment by alinask** [comment]
@yosefe @shamisp @bbenton Updated. Please review.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4890/ for details (Mellanox internal link).

File: src/uct/api/uct.h [hunk 2]
```diff
@@ -1318,6 +1313,16 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
  *                               remote peer.
  * @param [in]  length           Length of the private data.
  * @param [out] ep_p             Handle to the created endpoint.
+ *
+ * @return UCS_OK              - The connection to the server, on the client side,
+ *                               was established. No reply from the server is
+ *                               required and therefore the reply_cb callback
+ *                               won't be invoked.
+ *         UCS_INPROGRESS      - The connection to the remote peer was initiated.
```
Comment (bbenton) [inline src/uct/api/uct.h]: pls add the doxygen \return command to force to a new line, as well as ::<Name>: to format the reference, e.g.,
  \return ::UCS_INPROGRESS <desctiption>

File: src/uct/api/uct.h [hunk 3]
```diff
@@ -1318,6 +1313,16 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
  *                               remote peer.
  * @param [in]  length           Length of the private data.
  * @param [out] ep_p             Handle to the created endpoint.
+ *
+ * @return UCS_OK              - The connection to the server, on the client side,
```
Comment (bbenton) [inline src/uct/api/uct.h]: changing UCS_OK to ::UCS_OK results in a linkable reference

File: src/uct/api/uct.h [hunk 4]
```diff
@@ -1318,6 +1313,16 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
...
+ *                               was established. No reply from the server is
+ *                               required and therefore the reply_cb callback
+ *                               won't be invoked.
+ *         UCS_INPROGRESS      - The connection to the remote peer was initiated.
+ *                               The user will be notified of the connection
+ *                               establishment, on the client side, when the reply_cb
+ *                               callback will be invoked.
+ *         error code          - In case of an error.
```
Comment (bbenton) [inline src/uct/api/uct.h]: Suggest modifying to:
  \return \ref ucs_status_t "Error code" - In case of issues.

**Comment by alinask** [comment]
@bbenton Thanks for your comments!
It seems that none of documented return statuses has a linkable reference (::UCS_OK) in this file. 
In order not to change the appearance of only this function's doc, I kept it the way it was, so that it would look the same as the other functions. If you think that all the return statuses of all the functions should have a linkable reference, IMHO it should be done in a separate PR.
I applied the rest of the comments.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4917/ for details (Mellanox internal link).

**Comment by bbenton** [comment]
@alinask I got the idea for this from the way that the returns for the function  uct_iface_event_arm() were handled in uct.h.  But, I agree, this is a broader consistency issue which can be addressed through documentation cleanup.

**Comment by bbenton** [comment]
:+1: for doc/API

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4930/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4931/ for details (Mellanox internal link).

**Pull request merged**